breathecode/payments/views.py

```diff
@@ -996,6 +995,36 @@ def get(self, request, subscription_id=None, academy_id=None):
 
         return handler.response(serializer.data)
 
+    def put(self, request, subscription_id, academy_id=None):
+        lang = get_user_language(request)
+
+        if not (subscription := Subscription.objects.filter(id=subscription_id).first()):
+            raise ValidationException(
+                translation(lang, en="Subscription not found", es="No existe la suscripción", slug="not-found"),
+                code=404,
+            )
+
+        valid_statuses = [choice[0] for choice in Subscription._meta.get_field("status").choices]
+
+        allowed_fields = ["status", "valid_until", "plan"]
+        for field, value in request.data.items():
+            if field == "status" and value not in valid_statuses:
+                raise ValidationException(
+                    translation(
+                        lang,
+                        en=f"{field}: '{value}' is not a valid choice.",
+                        es=f"{field}: '{value}' no es una opción válida.",
+                        slug="invalid-choice",
+                    ),
+                    code=400,
+                )
+            if field in allowed_fields:
+                setattr(subscription, field, value)
+
+        subscription.save()
```

Comment:
With subscriptions Tomas could be referring to plan financings, double check this